raiden/tests/integration/test_send_queued_messages.py

```diff
@@ -0,0 +1,98 @@
+import pytest
+import gevent
+from raiden import waiting
+from raiden.transfer import views
+from raiden.tests.utils.network import CHAIN
+from raiden.tests.utils.transfer import (
+    direct_transfer,
+    assert_synced_channel_state,
+)
+from raiden.network.transport import MatrixTransport
+from raiden.app import App
+
+
+@pytest.mark.parametrize('in_memory_database', [False])
+@pytest.mark.parametrize('deposit', [10])
+@pytest.mark.parametrize('channels_per_node', [CHAIN])
+@pytest.mark.parametrize('number_of_nodes', [2])
+def test_recovery_happy_case(
+        raiden_network,
+        number_of_nodes,
+        deposit,
+        token_addresses,
+        network_wait,
+):
+    app0, app1 = raiden_network
+    token_address = token_addresses[0]
+    chain_state = views.state_from_app(app0)
+    payment_network_id = app0.raiden.default_registry.address
+    token_network_identifier = views.get_token_network_identifier_by_token_address(
+        chain_state,
+        payment_network_id,
+        token_address,
+    )
+
+    # stop app1 - direct transfer must be left unconfirmed
+    app1.stop()
+
+    # make a few transfers from app0 to app2
+    amount = 1
+    spent_amount = 7
+    for _ in range(spent_amount):
+        direct_transfer(
+            app0,
+            app1,
+            token_network_identifier,
+            amount,
+            timeout=network_wait * number_of_nodes,
+        )
+
+    # restart app0
+    app0.raiden.stop()
+
+    new_transport = MatrixTransport(
+        app0.raiden.config['transport']['matrix'],
+    )
+
+    app0_restart = App(
+        config=app0.config,
+        chain=app0.raiden.chain,
+        query_start_block=0,
+        default_registry=app0.raiden.default_registry,
+        default_secret_registry=app0.raiden.default_secret_registry,
+        transport=new_transport,
+        discovery=app0.raiden.discovery,
+    )
+
+    app0.stop()
+    del app0  # from here on the app0_restart should be used
+    app1.start()
+
+    app0_restart.start()
+
+    waiting.wait_for_healthy(
+        app0_restart.raiden,
+        app1.raiden.address,
+        network_wait,
+    )
+    waiting.wait_for_healthy(
+        app1.raiden,
+        app0_restart.raiden.address,
+        network_wait,
+    )
+
+    exception = RuntimeError('Timeout while waiting for new channel')
+    with gevent.Timeout(5, exception=exception):
+        waiting.wait_for_newchannel(
```

Comment:
humm, which new channel?

Reply:
it makes sure the internal channel state is actually synced...

Reply:
> it makes sure the internal channel state is actually synced...

What do you mean by synced? that both nodes have the same consistent state? or that the node knows about the opened channel?

Because for the first, this is not useful, for the second it doesn't hurt